File: app/src/main/java/io/github/ovso/blackbox/ui/base/view/BaseActivity.java
package io.github.ovso.blackbox.ui.base.view;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;
import android.view.ViewGroup;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;
import dagger.android.support.DaggerAppCompatActivity;
import io.github.ovso.blackbox.R;
import io.github.ovso.blackbox.main.base.view.MyAdView;

public abstract class BaseActivity extends DaggerAppCompatActivity {
  protected @BindView(R.id.toolbar) Toolbar toolbar;
  private Unbinder bind;
  @BindView(R.id.ad_container) ViewGroup adContainer;
  @Override protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(getLayoutResID());
    bind = ButterKnife.bind(this);
    setSupportActionBar(toolbar);
    getSupportActionBar().setDisplayShowTitleEnabled(isTitle());
    onCreated(savedInstanceState);
    adContainer.addView(MyAdView.getAdmobAdView(this));
  }

  protected abstract int getLayoutResID();

  protected abstract void onCreated(@Nullable Bundle savedInstanceState);

  @Override public boolean onOptionsItemSelected(MenuItem item) {
    finish();
    return super.onOptionsItemSelected(item);
  }

  @Override protected void onDestroy() {
    super.onDestroy();
    if (bind != null) {
      bind.unbind();
    }
  }

  public boolean isTitle() {
    return true;
  }
}

File: app/src/main/java/io/github/ovso/blackbox/ui/base/view/BaseActivity.kt
package io.github.ovso.blackbox.ui.base.view

import android.os.Bundle
import android.view.MenuItem
import android.view.ViewGroup
import dagger.android.support.DaggerAppCompatActivity
import io.github.ovso.blackbox.Ads
import io.github.ovso.blackbox.R
import io.github.ovso.blackbox.exts.loadAdaptiveBanner
import kotlinx.android.synthetic.main.app_bar_main.*

abstract class BaseActivity : DaggerAppCompatActivity() {

  protected abstract val layoutResID: Int

  private val isTitle: Boolean
    get() = true

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(layoutResID)
//    setSupportActionBar(toolbar)
//    supportActionBar!!.setDisplayShowTitleEnabled(isTitle)
    onCreated(savedInstanceState)
    findViewById<ViewGroup>(R.id.ad_container)?.let {
      loadAdaptiveBanner(it, Ads.BANNER_UNIT_ID)
    }

  }

  protected abstract fun onCreated(savedInstanceState: Bundle?)

  override fun onOptionsItemSelected(item: MenuItem): Boolean {
    finish()
    return super.onOptionsItemSelected(item)
  }
}
